### ISSUE TITLE
detect possible option injection

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#2305]
Please detect possible option injection. This is when variables or wildcards lead to passing strings to commands that are then interpreted as options for those commands. This can result in incorrect behaviour from commands or in some cases arbitrary command execution. The canonical security paper on this is [Back To The Future: Unix Wildcards Gone Wild](http://www.defensecode.com/public/DefenseCode_Unix_WildCards_Gone_Wild.txt) by Leon Juranic. The Bash Pitfalls wiki page also [describes](https://mywiki.wooledge.org/BashPitfalls#Filenames_with_leading_dashes) this problem.

#### For new checks and feature suggestions

- [x] I was unable to check https://www.shellcheck.net/ because it requires JavaScript
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
cp "$1" "$2"
for f in * ; do
    cp "$f" /target
done
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

shellcheck should warn that options may be injected into the cp commands.

#### Here are three different correct ways to do things:

```sh
#!/bin/sh
cp -- "$1" "$2"
for f in ./* ; do
    cp "$f" /target
done
```

```sh
#!/bin/sh
cp -- "$1" "$2"
for f in * ; do
    cp -- "$f" /target
done
```

```sh
#!/bin/sh
cp -- "$1" "$2"
for f in * ; do
    cp "./$f" /target
done
```

**Comment by pabs3** [comment]
I just found that [SC2035](https://www.shellcheck.net/wiki/SC2035) exists to check that globs don't become options, but I note that it does not get triggered by the above code. So probably SC2035 needs to get expanded to cover the above cases.
